wordpress-mobile/WordPress-Android
# Adding liker faces train in reader and allow to access the list of likers

Part of the #14138. This PR shows liker faces (up to 5) in the Reader Post Details and allow to open the list of likers already used in the Notifications section (see #14361).

| ![image](https://user-images.githubusercontent.com/47797566/113787546-8844b180-973b-11eb-95b2-f4d1f307c818.png) | ![image](https://user-images.githubusercontent.com/47797566/113787584-94307380-973b-11eb-899c-0fe897bb87d4.png) |
|---|---|

Hey @mattmiklic 👋. I'm still looking a couple of improvements code-wise but this should be ready for a design review (as for the previous one please note the `LikesEnhancementsFeatureConfig`  feature flag must be activated using the CI built apk 🙇 ). Want to highlight a deviation from original design spec. Unless I'm doing something wrong, it seems to me I'm obtaining a result more similar to spec using a spacing of -4dp (the apk in here is using this spacing) instead of -2dp (as per spec). Below the result with -2dp (top image) and -4dp (bottom image)

<p align=center><image src="https://user-images.githubusercontent.com/47797566/113186945-8d47c380-9258-11eb-97ca-34669ebc7203.png" width=300/></p>


## To test

The feature is behind a Feature Flag to which we are adding in next PRs. Good to compare the test below side by side with a develop installation.

- Try to open a post in the reader and check the like faces are loaded (notice the initial loading state)
- Check number of likers are consistent with the number next to the like star
- Check you can tap on the faces row to get the list of likers (this should be consistent with the list in the previous notifications PR)
- Try to like/unlike and check you get the number updated
- Try to make the above checks both with posts with lot of likes and under 5 likes or no likes
- Use a super short post to be able to check the loading phases of like  section without scrolling the post webview
- Check screen rotation states
- When having issues loading likers data (for example setting air mode) check that you have:   
    - if no likers data were loaded before -> you get an empty state error string
    - if some likers data were loaded before -> you get the rendered cached data and a snackbar with the issue message
- Check dark mode is consistent
- With feature flag off, smoke test to check that we are having same behaviour as before.

## Regression Notes
This feature is still behind a feature flag; regression notes will be highlighted as the feature parts are shipped to user. Also unit testing is being addressed in a dedicated PR.

PR submission checklist:
